js/game.js

```diff
@@ -29,32 +32,104 @@ export default class Game {
       )
         this.tamagotchi.health.value--;
       this.tamagotchi.displayHealth(this.tamagotchi.health.element);
+      this.tamagotchi.updateState();
     }
   };
 
   decreaseHunger = () => {
+    this.tamagotchi.checkState();
     if (this.tamagotchi.hunger.value > 0) this.tamagotchi.hunger.value--;
     this.tamagotchi.displayHunger(this.tamagotchi.hunger.element);
+    this.tamagotchi.updateState();
   };
 
   decreaseEnergy = () => {
+    this.tamagotchi.checkState();
     if (this.tamagotchi.energy.value > 0) {
       this.tamagotchi.energy.value--;
-      if (this.tamagotchi.fun.value <= 0) this.tamagotchi.energy.value--;
+      if (this.tamagotchi.fun.value <= 0 && this.tamagotchi.energy.value > 0)
+        this.tamagotchi.energy.value--;
     }
     this.tamagotchi.displayEnergy(this.tamagotchi.energy.element);
+    this.tamagotchi.updateState();
   };
 
   decreaseFun = () => {
+    this.tamagotchi.checkState();
     if (this.tamagotchi.fun.value > 0) this.tamagotchi.fun.value--;
     this.tamagotchi.displayFun(this.tamagotchi.fun.element);
+    this.tamagotchi.updateState();
   };
 
   end = () => {
     clearInterval(this.healthIntervalId);
     clearInterval(this.hungerIntervalId);
     clearInterval(this.energyIntervalId);
     clearInterval(this.funIntervalId);
-    console.log("Game over");
+  };
+
+  resume = () => {
+    this.healthIntervalId = setInterval(this.decreaseHealth, 1000);
+    this.hungerIntervalId = setInterval(this.decreaseHunger, 1000);
+    this.energyIntervalId = setInterval(this.decreaseEnergy, 2000);
+    this.funIntervalId = setInterval(this.decreaseFun, 1000);
+  };
+
+  eatBtn = document.getElementById("eatBtn");
```

Comment:
**nitpick:** We should not have hardcoded values here. Check your `app.js` file - you already pass some values there. Think of it as a configuration file where you declare all needed values from single entry point. After that you should be able to use these values inside your class logic

Reply:
I managed passing buttons through app.js but when I tried to do the same thing wioth nimo and petState elements it stopped working properly. I've tried to pass it in all the ways that I know but I failed, so I vote for leaving it as it is. 😃.